packages/serverpod/lib/src/server/websocket_request_handlers/endpoint_websocket_request_handler.dart

```diff
@@ -139,37 +126,41 @@ abstract class EndpointWebsocketRequestHandler {
     }
   }
 
-  static Future<void> _callStreamOpened(
+  static Future<void> _callStreamOpenedOnAll(
     StreamingSession session,
-    Endpoint endpoint,
+    EndpointDispatch endpointDispatch,
   ) async {
-    try {
-      session.endpoint = endpoint.name;
-      var authFailed = await EndpointDispatch.canUserAccessEndpoint(
-        () => session.authenticated,
-        endpoint.requireLogin,
-        endpoint.requiredScopes,
-      );
-      if (authFailed == null) await endpoint.streamOpened(session);
-    } catch (e) {
-      return;
+    for (var endpointConnector in endpointDispatch.connectors.values) {
+      EndpointConnector connector;
+      try {
+        session.endpoint = endpointConnector.name;
+        connector = await endpointDispatch.getAuthorizedEndpoint(
+            session: session, endpointPath: endpointConnector.name);
+        await connector.endpoint.streamOpened(session);
+      } on NotAuthorizedException {
+        continue;
+      } catch (e) {
+        return;
+      }
     }
   }
 
-  static Future<void> _callStreamClosed(
+  static Future<void> _callStreamClosedOnAll(
     StreamingSession session,
-    Endpoint endpoint,
+    EndpointDispatch endpointDispatch,
   ) async {
-    try {
-      session.endpoint = endpoint.name;
-      var authFailed = await EndpointDispatch.canUserAccessEndpoint(
-        () => session.authenticated,
-        endpoint.requireLogin,
-        endpoint.requiredScopes,
-      );
-      if (authFailed == null) await endpoint.streamClosed(session);
-    } catch (e) {
-      return;
+    for (var endpointConnector in endpointDispatch.connectors.values) {
+      EndpointConnector connector;
+      try {
+        session.endpoint = endpointConnector.name;
+        connector = await endpointDispatch.getAuthorizedEndpoint(
+            session: session, endpointPath: endpointConnector.name);
+        await connector.endpoint.streamClosed(session);
+      } on NotAuthorizedException {
+        continue;
+      } catch (e) {
+        return;
+      }
```

Comment:
Suggestion: Move loop out to the outside again.

From my perspective, having the loop on the outside more clearly communicated what was happening.
There is also a slight logical change with this implementation where we would call `return` for all endpoints in the loop if another exception happens as opposed to only skipping a single entry.

Reply:
> we would call return for all endpoints in the loop if another exception happens as opposed to only skipping a single entry.

Maybe I'm missing something, but I don't think this is true? It seems to be the same behavior in both cases.

That said, I do agree that it's more explicit to have the loop outside. The downside was that the function signature needed more arguments in that case and looked a bit ugly, so I weighing the two options I thought this one was best. Lets discuss which one you prefer!

Reply:
[8105f19](https://github.com/serverpod/serverpod/pull/2676/commits/8105f196c352927da6fbdd3c8c1318a60cb2a981)